framework/test/unit/node/endpoint.spec.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { Event, StateStore } from '@liskhq/lisk-chain';
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { InMemoryKVStore } from '@liskhq/lisk-db';
+import { ModuleEndpointContext } from '../../../src';
+import { Endpoint } from '../../../src/node/endpoint';
+import { createImmutableAPIContext } from '../../../src/node/state_machine';
+import { nodeConfig } from '../../utils/configs';
+import { fakeLogger } from '../../utils/node';
+
+const createContext = (
```

Comment:
can we use some existing createContext funcs in testing or add this one to make it reusable